Sources/libhostmgr/Model/EnvFileParser.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+
+struct EnvFile {
+    var configuration: [String: String]
+
+    static func from(_ url: URL) throws -> EnvFile {
+        try EnvFileParser(string: String(contentsOf: url)).parse()
+    }
+
+    static func from(_ string: String) throws -> EnvFile {
+        try EnvFileParser(string: string).parse()
+    }
+
+    subscript(key: String) -> String? {
+        configuration[key]
+    }
+
+    /// Apply the values in the `.env` file to the environment
+    func apply() {
+        var environmentCopy = ProcessInfo.processInfo.environment
+        for (key, value) in self.configuration {
+            environmentCopy[key] = value
+        }
+        Process().environment = environmentCopy
+    }
+}
+
+class EnvFileParser {
+
+    private let string: String
+
+    init(string: String) {
+        self.string = string
+    }
+
+    func parse() throws -> EnvFile {
+        let pairs = try self.string
+            .components(separatedBy: "\n")
+            .compactMap(convertLineToKeyValuePair)
+            .reduce(into: [String: String]()) { $0[$1.0] = $1.1 }
+
+        return EnvFile(configuration: pairs)
+    }
+
+    func convertLineToKeyValuePair(_ line: String) throws -> (String, String)? {
+        let trimmedLine = line.trimmingWhitespace
+        if trimmedLine.isEmpty {
+            return nil
+        }
+
+        guard let separatorIndex = line.firstIndex(of: "=") else {
+            return nil
+        }
+
+        let startOfValueIndex = line.index(separatorIndex, offsetBy: 1)
+
+        let key = String(line[..<separatorIndex]).trimmingWhitespace
+        let value = String(line[startOfValueIndex...]).trimmingWhitespace
+
+        if let quotedValue = try extractQuotedPortionOfValue(value) {
+            return (key, quotedValue)
+        }
+
+        if let indexOfCommentMarker = value.firstIndex(of: "#") {
+            return (key, String(value[..<indexOfCommentMarker]).trimmingWhitespace)
+        }
+
+        return (key, value)
+    }
+
+    func extractQuotedPortionOfValue(_ value: String) throws -> String? {
+        let expression = try NSRegularExpression(pattern: #"^"(?<string>.*)""#)
```

Comment:
It appears this regex doesn't support single quote, which is supported in [bkeepers/dotenv (the ruby implementation)](https://github.com/bkeepers/dotenv/blob/v2.8.1/spec/dotenv/parser_spec.rb#L41-L43) and [motdotla/dotenv (the nodejs implementation)](https://github.com/motdotla/dotenv/blob/v16.0.3/tests/.env-multiline#L6).

I'm wondering what's the thinking behind implementing the "dotenv convention" (is there a standard for it?) ourselves, rather than using a third party solution which appears having the same set of parsing rules as motdotla/dotenv (the nodejs implementation)? My worry is, missing some of the convention in our own implementation might introduce subtle bugs when dealing with a env file that's not created by us (i.e. buildkite).